Widgets/Bars/ClassBar.lua

```diff
@@ -249,7 +262,9 @@ local function TogglePowerEvents(self, enable)
 end
 
 ---@param self ClassBarWidget
-local function ShouldShow(self)
+---@param event ("UNIT_ENTERED_VEHICLE"|"UNIT_EXITED_VEHICLE"|"PLAYER_SPECIALIZATION_CHANGED"|"UNIT_DISPLAYPOWER"|"PLAYER_REGEN_ENABLED"|"PLAYER_REGEN_DISABLED")?
+local function ShouldShow(self, event)
+    if self.hideOutOfCombat and not (UnitAffectingCombat("player") or event == "PLAYER_REGEN_DISABLED") then return false end
```

Comment:
Are you certain that is the proper event? You want a falsy return outside of combat, which is triggered by the event "PLAYER_REGEN_ENABLED".

Also the `or` cnd after `and` needs to be wrapped.
```suggestion
    if self.hideOutOfCombat and (not (UnitAffectingCombat("player") or event == "PLAYER_REGEN_DISABLED")) then return false end
```

Reply:
I will double check.

Reply:
I think you're right. I split up the conditions to be a bit more verbose and hopefully less confusing.